app-server/app/src/main/scala/user/User.scala

```diff
@@ -20,14 +47,37 @@ object UserService {
 
     database.db.run {
       Users.filter(_.id === id).result.headOption
+    } map {
+      userOption => userOption match {
+        case Some(user) => new UsersRowApi(user)
+        case _ => {
+          // catch this to return 404
+          throw new Exception("404 Resource not found")
+        }
+      }
     }
   }
 
-  def createUser(user:UsersRow)(implicit database:Database, ec:ExecutionContext) = {
+  def createUser(user:UsersRowCreate)(implicit database:Database, ec:ExecutionContext) = {
     import database.driver.api._
+    val now = new Timestamp((new java.util.Date()).getTime())
+    val newUUID = java.util.UUID.randomUUID
+    val userRow = UsersRow.apply(
+      id=newUUID,
+      createdAt=now,
+      modifiedAt=now,
+      isActive=user.isActive,
+      isStaff=user.isStaff,
+      email=user.email,
+      firstName=user.firstName,
+      lastName=user.lastName,
+      organizationId=user.organizationId
+    )
 
     database.db.run {
-      Users.forceInsert(user)
+      Users.forceInsert(userRow)
+    } map {
+      x => new UsersRowApi(userRow)
```

Comment:
Why doesn't this return a `UsersRow`? Read only fields shouldn't be updateable, but they should be able to be read.